upstairs/src/dummy_downstairs_tests.rs

```diff
@@ -1,9 +1,9 @@
 // Copyright 2023 Oxide Computer Company
 
-#[cfg(test)]
+#[cfg(not(test))]
+compile_error!("dummy_downstairs should only be used in unit tests");
+
 pub(crate) mod protocol_test {
```

Comment:
After the PR is approved, I'll push a final (purely mechanical) commit which removes `mod protocol_test`, saving a level of indentation.